controller/controller_server/csi_controller_server.py

```diff
@@ -387,9 +387,34 @@ def ControllerUnpublishVolume(self, request, context):
 
     def ValidateVolumeCapabilities(self, request, context):
         logger.info("ValidateVolumeCapabilities")
-        context.set_code(grpc.StatusCode.UNIMPLEMENTED)
-        logger.info("finished ValidateVolumeCapabilities")
-        return csi_pb2.ValidateVolumeCapabilitiesResponse()
+        try:
+
+            utils.validate_validate_volume_capabilities_request(request)
+
+            secrets = request.secrets
+            user, password, array_addresses = utils.get_array_connection_info_from_secret(secrets)
+
+            array_type, volume_id = utils.get_volume_id_info(request.volume_id)
+
+            with get_agent(user, password, array_addresses, array_type).get_mediator() as array_mediator:
+                volume = array_mediator.get_object_by_id(object_id=volume_id, object_type=config.VOLUME_TYPE_NAME)
+
+            if request.volume_context:
+                utils.validate_volume_context_match_volume(request.volume_context, volume)
+            logger.info("finished ValidateVolumeCapabilities")
+            return utils.generate_csi_validate_volume_capabilities_response(request.volume_context,
+                                                                            request.volume_capabilities,
+                                                                            request.parameters)
```

Comment:
according to the [spec](https://github.com/container-storage-interface/spec/blob/v1.3.0/spec.md#validatevolumecapabilities), this is:
```
The volume creation parameters validated by the plugin.
```
could you point me to where this is validated?

Reply:
its in now

Reply:
I think the spec wants us to validate whether or not the parameters in the request match the existing volume parameters.
what do you think? should we just ensure there is some non-empty pool value?

Reply:
this parameter field is the values transferred by the storage class, except of pool and  SpaceEfficiency we are not expecting any other values

Reply:
done

Reply:
1. `except of pool and SpaceEfficiency we are not expecting any other values` - please add a link here that supports this claim (I want to make sure we did not miss any other parameter).
2. as I wrote, `I think the spec wants us to validate whether or not the parameters in the request match the existing volume parameters.` - I couldn't find where this is `done`. do you disagree? if not, please add a link here `to where this is validated (+ unit test for it)`.

Reply:
we have 4 official parameters:
[space_efficiency, pool, prefix](https://github.com/IBM/ibm-block-csi-driver/blob/a9950ee1c66276876951b1aa629725ef34ca7cc3/controller/controller_server/utils.py#L427)
[fs_type](https://github.com/IBM/ibm-block-csi-driver/blob/a9950ee1c66276876951b1aa629725ef34ca7cc3/controller/controller_server/utils.py#L51)

Reply:
nice coverage!